app/services/themes/destroy.rb

```diff
@@ -28,6 +28,7 @@ class Themes::Destroy
   end
 
   model :theme
+  policy :not_default
```

Comment:
and where to we check it's not system?

Reply:
That one is checked here:

https://github.com/discourse/discourse/blob/main/app/services/themes/destroy.rb#L27